### ISSUE TITLE
[Bug] "ERROR": "Platform bambu_lab does not generate unique IDs"

### ISSUE THREAD

**Issue by leonklingele** [greghesp/ha-bambulab#145]
### Describe the bug

HA issues the following lines every second:

```
homeassistant    | 2023-05-15 16:33:16.339 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_ExternalSpool_external_spool already exists - ignoring sensor.x1c_ID_externalspool_external_spool
homeassistant    | 2023-05-15 16:33:16.344 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID1_humidity_index already exists - ignoring sensor.x1c_ID_ams_1_humidity_index_2
homeassistant    | 2023-05-15 16:33:16.348 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID2_humidity_index already exists - ignoring sensor.x1c_ID_ams_1_humidity_index
homeassistant    | 2023-05-15 16:33:16.352 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID1_humidity already exists - ignoring sensor.x1c_ID_ams_1_humidity_2
homeassistant    | 2023-05-15 16:33:16.357 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID2_humidity already exists - ignoring sensor.x1c_ID_ams_1_humidity
homeassistant    | 2023-05-15 16:33:16.362 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID1_temperature already exists - ignoring sensor.x1c_ID_ams_1_temperature_2
homeassistant    | 2023-05-15 16:33:16.367 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID2_temperature already exists - ignoring sensor.x1c_ID_ams_1_temperature
homeassistant    | 2023-05-15 16:33:16.371 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID1_tray_1 already exists - ignoring sensor.x1c_ID_ams_1_tray_1_2
homeassistant    | 2023-05-15 16:33:16.376 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID2_tray_1 already exists - ignoring sensor.x1c_ID_ams_1_tray_1
homeassistant    | 2023-05-15 16:33:16.381 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID1_tray_2 already exists - ignoring sensor.x1c_ID_ams_1_tray_2_2
homeassistant    | 2023-05-15 16:33:16.386 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID2_tray_2 already exists - ignoring sensor.x1c_ID_ams_1_tray_2
homeassistant    | 2023-05-15 16:33:16.390 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID1_tray_3 already exists - ignoring sensor.x1c_ID_ams_1_tray_3_2
homeassistant    | 2023-05-15 16:33:16.395 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID2_tray_3 already exists - ignoring sensor.x1c_ID_ams_1_tray_3
homeassistant    | 2023-05-15 16:33:16.399 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID1_tray_4 already exists - ignoring sensor.x1c_ID_ams_1_tray_4_2
homeassistant    | 2023-05-15 16:33:16.404 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID X1C_ID_AMS_ID2_tray_4 already exists - ignoring sensor.x1c_ID_ams_1_tray_4
homeassistant    | 2023-05-15 16:33:16.408 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID ID_wifi_signal already exists - ignoring sensor.x1_ID_wi_fi_signal
homeassistant    | 2023-05-15 16:33:16.412 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID ID_bed_temp already exists - ignoring sensor.x1_ID_bed_temperature
homeassistant    | 2023-05-15 16:33:16.417 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID ID_target_bed_temp already exists - ignoring sensor.x1_ID_target_bed_temperature
homeassistant    | 2023-05-15 16:33:16.421 ERROR (MainThread) [homeassistant.components.sensor] Platform bambu_lab does not generate unique IDs. ID ID_chamber_temp already exists - ignoring sensor.x1_ID_chamber_temperature
```

### To Reproduce

See Home Assistant logs.

### Expected Behaviour

No such line should be logged.

### What device are you using?

X1C

### Home Assistant Version

Home Assistant 2023.3.6

### Component Version

v1.3.1

**Comment by davidt94** [comment]
Also seeing the same issue.

Logger: homeassistant.components.sensor
Source: helpers/entity_platform.py:554
Integration: Sensor (documentation, issues)
First occurred: 1:13:49 AM (60 occurrences)
Last logged: 1:13:57 AM

Platform bambu_lab does not generate unique IDs. ID 00M00A2C0604266_start_time already exists - ignoring sensor.bambu_lab_x1_carbon_start_time
Platform bambu_lab does not generate unique IDs. ID 00M00A2C0604266_remaining_time already exists - ignoring sensor.bambu_lab_x1_carbon_remaining_time
Platform bambu_lab does not generate unique IDs. ID 00M00A2C0604266_end_time already exists - ignoring sensor.bambu_lab_x1_carbon_end_time
Platform bambu_lab does not generate unique IDs. ID 00M00A2C0604266_current_layer already exists - ignoring sensor.bambu_lab_x1_carbon_current_layer
Platform bambu_lab does not generate unique IDs. ID 00M00A2C0604266_total_layers already exists - ignoring sensor.bambu_lab_x1_carbon_total_layer_count

**Comment by AdrianGarside** [comment]
Looks like you renamed your devices? I'm trying to track down what's going wrong when you do that - so far I haven't been able to reproduce the problem.

**Comment by leonklingele** [comment]
> Looks like you renamed your devices?

No, I didn't, but for whatever reason I have these 5 devices available:

- X1C_ID
- X1C_ID_AMS_1
- X1C_ID_AMS_2
- X1C_ID_AMS_2
- X1C_ID_ExternalSpool

Two AMS are connected to the printer. I don't know why the "AMS_2" is shown twice.

I moved one of them to the "Living Room" area, might this be related to the issue?

<img width="836" alt="image" src="https://github.com/greghesp/ha-bambulab/assets/5585491/fdd34e8a-7abf-4cb2-9caf-fb765b35a6c6">

The one no inside the "Living Room" shows "Unavailable" on all sensor values.

**Comment by AdrianGarside** [comment]
Thanks. I think I’m beginning to understand what might be causing this and several of the other issues.

**Comment by mydevpeeps** [comment]
I think this is happening because of the duplicate AMS issue....

**Comment by AdrianGarside** [comment]
Yes. I’m fairly sure they are the same bug. It’s a race condition but I don’t have a way to guarantee I hit the bad case so when I make a potential fix I might not be able to confirm it actually helps.

**Comment by AdrianGarside** [comment]
Possible fix for this in 1.4.2-dev release.

**Comment by Scoop2389** [comment]
This issue is still present for me, but only occurs whenever home assistant is started.

Logs:
```
Platform bambu_lab does not generate unique IDs. ID [REDACTED]_total_layers already exists - ignoring sensor.x1c_[REDACTED]_total_layer_count
Platform bambu_lab does not generate unique IDs. ID [REDACTED]_tray_now already exists - ignoring sensor.x1c_[REDACTED]_active_tray
Platform bambu_lab does not generate unique IDs. ID [REDACTED]_gcode_file already exists - ignoring sensor.x1c_[REDACTED]_gcode_filename
Platform bambu_lab does not generate unique IDs. ID [REDACTED]_subtask_name already exists - ignoring sensor.x1c_[REDACTED]_task_name
Platform bambu_lab does not generate unique IDs. ID [REDACTED]_active_tray already exists - ignoring sensor.x1c_[REDACTED]_active_tray_2
```
